x-pack/plugin/sql/src/test/java/org/elasticsearch/xpack/sql/optimizer/OptimizerTests.java

```diff
@@ -1023,4 +1022,43 @@ public void testReplaceAttributesWithTarget() {
         gt = (GreaterThan) and.left();
         assertEquals(a, gt.left());
     }
+
+    //
+    // ReplaceSumWithStats rule
+    //
+    public void testSumIsReplacedWithStats() {
+        FieldAttribute fa = getFieldAttribute();
+        Sum sum = new Sum(EMPTY, fa);
+        
+        Alias sumAlias = new Alias(EMPTY, "sum", sum);
+        
+        Aggregate aggregate = new Aggregate(EMPTY, FROM(), emptyList(), asList(sumAlias));
+        LogicalPlan optimizedPlan = new Optimizer().optimize(aggregate);
+        assertTrue(optimizedPlan instanceof Aggregate);
+        Aggregate p = (Aggregate) optimizedPlan; 
+        assertEquals(1, p.aggregates().size());
+        assertTrue(p.aggregates().get(0) instanceof Alias);
+        Alias alias = (Alias) p.aggregates().get(0);
+        assertTrue(alias.child() instanceof InnerAggregate);
+        assertEquals(sum, ((InnerAggregate) alias.child()).inner());
+    }
+
+    /**
+     * Once the bug is fixed, the above {@link OptimizerTests#testSumIsReplacedWithStats()} should be
+     * invalid and should be deleted, and the test below should apply.
+     */
+    @AwaitsFix(bugUrl = "https://github.com/elastic/elasticsearch/issues/45251")
```

Comment:
Incorrect issue referenced - this PR provides the workaround in SQL for ES. Once the issue in ES gets solved, this test can be enabled.